lib/mg.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+class Mg {
+  constructor(args, ops = {archive: false}) {
+    this._getPattern = this._getPattern.bind(this);
+    this._haveFlags = this._haveFlags.bind(this);
+    this._flagChars = this._flagChars.bind(this);
+    this._caseSensitive = this._caseSensitive.bind(this);
+    this._excludePattern = this._excludePattern.bind(this);
+    this._directory = this._directory.bind(this)
```

Comment:
I believe that you may delete all of these lines binding `this` by creating each of the functions using arrow functions. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this

Reply:
Hmm. I agree that the code here feels messy and should be tidied 👍 

I'm not sure how to implement your solution though, each of these functions are class methods and I can't see how to use arrow functions to fix the problem.

I have gone about this via the "Caveman Mode" solution from here: https://ponyfoo.com/articles/binding-methods-to-class-instance-objects#caveman-mode.

I'll have a think about how this code can be cleaned up.

Reply:
I can't think of a nicer way to do this. I don't want to add the [`auto-bind`](https://github.com/sindresorhus/auto-bind) dependency from the article.